scipy/stats/tests/test_stats.py

```diff
@@ -538,6 +538,89 @@ def test_gh3014(self):
         odds, pvalue = stats.fisher_exact([[1, 2], [9, 84419233]])
 
 
+class TestBarnardExact(object):
+    """Some tests to show that barnard_exact() works correctly.
+    """
+
+    def test_basic(self):
+        # results test against R's Barnard's Unconditional Test
+
+        barnard_exact = stats.barnard_exact
+        TX_obs, pvalue = barnard_exact([[7, 12], [8, 3]])
+        assert_almost_equal(
+            [TX_obs, pvalue],
+            [1.89433807606, 0.0681534327315], decimal=5)
+
+        TX_obs, pvalue = barnard_exact(
+            [[43, 40], [10, 39]])
+        assert_almost_equal(
+            [TX_obs, pvalue],
+            [-3.55541, 0.000362832], decimal=5)
+
+        res = barnard_exact([[5, 1], [10, 10]])[1]
+        assert_approx_equal(res, 0.1562775463060, significant=4)
+        res = barnard_exact([[5, 15], [20, 20]])[1]
+        assert_approx_equal(res, 0.0663635014149, significant=4)
+        res = barnard_exact([[5, 16], [20, 25]])[1]
+        assert_approx_equal(res, 0.1169848521762, significant=4)
+        res = barnard_exact([[10, 5], [10, 1]])[1]
+        assert_approx_equal(res, 0.177536588912, significant=4)
+        res = barnard_exact([[5, 0], [1, 4]])[1]
+        assert_approx_equal(res, 0.01367187500000, significant=4)
+        res = barnard_exact([[0, 1], [3, 2]])[1]
+        assert_approx_equal(res, 0.509667991865, significant=4)
+        res = barnard_exact([[0, 2], [6, 4]])[1]
+        assert_approx_equal(res, 0.1970196187397, significant=4)
+        res = barnard_exact([[2, 7], [8, 2]])
+        assert_approx_equal(res[1], 0.01921081542969, significant=4)
+        assert_approx_equal(res[0], 2.51847494516, significant=4)
+
+    def test_raises(self):
+        # test we raise an error for wrong input number of nuisances.
+        with assert_raises(ValueError):
```

Comment:
Please use the `match` keyword to test for the specific error message.

Reply:
Resolved in a5c581b9a67b15be3312284e9f0309ee6b0f7516.
Thank you !

Reply:
Thanks. For future comments, please let me click the "resolve" button, as I need to check that the comments have been addressed. I will try to be reasonably prompt about coming back to this and resolving them so they don't stay open forever if they're actually resolved. In the meantime, if you'd like, you're welcome to use the reaction symbols (like the one I applied to your comment here) to mark things you've addressed.

Reply:
Oh ok, I'll remember for the next time